### ISSUE TITLE
Confused entities with multiple amps/systems.

### ISSUE THREAD

**Issue by GabrielGoldsteinAnidea** [thebradleysanders/Monoprice-6-Zone-Home-Controller#7]
I've had a single socket:// setup working pretty well for a while.

Added another amp in a different location and setup a second socket:// device receiver.  All of the entities appear to have been created ok, however, the entities between the two systems are getting confused.

If I change an 'Area' on zone 11 on system B, it will change on system A.  I've also noticed that one of my automations which has been working well on system A now fails on occasion as it tries to adjust the volume on the parallel zone on system B.

1.  Is there a way I can configure this plugin some they don't overlap.  
2.  Is there a way I can get the names of the system (amp) to show up in the entitiy name?

**Comment by thebradleysanders** [comment]
Yes, I can see why this would be an issue. Should be an easy fix. I’ll have it out soon.Adding the name would separate the two in code also.Thanks for the feedback.Sent from my iPhoneOn Aug 3, 2023, at 3:50 PM, GabrielGoldsteinAnidea ***@***.***> wrote:﻿
I've had a single socket:// setup working pretty well for a while.
Added another amp in a different location and setup a second socket:// device receiver.  All of the entities appear to have been created ok, however, the entities between the two systems are getting confused.
If I change an 'Area' on zone 11 on system B, it will change on system A.  I've also noticed that one of my automations which has been working well on system A now fails on occasion as it tries to adjust the volume on the parallel zone on system B.

Is there a way I can configure this plugin some they don't overlap.
Is there a way I can get the names of the system (amp) to show up in the entitiy name?

**Comment by GabrielGoldsteinAnidea** [comment]
Any chance you can update this?

**Comment by eatonaaron** [comment]
any updates on this??

**Comment by GabrielGoldsteinAnidea** [comment]
I saw the dev tree.  Should I try that?  I dropped it in and HA wouldn't load it.  Do I need to reinstall it fresh and delete existing entities or should it drop right in?

**Comment by thebradleysanders** [comment]
You are correct, the dev branch is my attempt to fix the issue. However it is not yet working. As I have time I will continue to work on it. It been a challenging issue.

**Comment by GabrielGoldsteinAnidea** [comment]
This may help.  I loaded the native monoprice integration and it handles both devices fine.  Have you tried diffing your code to the latest HA integration to see how they handled it?  I've been meaning to try it myself, but I don't know how to do that yet.  Thanks!
